### PR TITLE
[ROX-17735] : Add ReportSnapshot data model

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

- Added ReportSnapshot proto autogenerated storage object
- Added postgres schema and store for ReportSnapshot
- Added datastore and searcher implementations

## Checklist
- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Unit tests

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: central/reports/snapshot/datastore/datastore.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	GetMany(ctx context.Context, ids []string) ([]*storage.ReportSnapshot, error)
+
+	AddReportSnapshot(ctx context.Context, report *storage.ReportSnapshot) error
+	DeleteReportSnapshot(ctx context.Context, id string) error
+
+	Walk(ctx context.Context, fn func(report *storage.ReportSnapshot) error) error
+}
+
+// New returns a new instance of a DataStore
+func New(storage pgStore.Store, searcher search.Searcher) (DataStore, error) {
```
Comment (md2119) [inline central/reports/snapshot/datastore/datastore.go:36]: error is nil always
change signature to return `DataStore` only.

File: proto/storage/report_snapshot.proto [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    string name = 2;
+}
+
+message NotifierSnapshot {
+    oneof notifier_config {
+        EmailNotifierSnapshot email_config = 1;
+    }
+}
+
+message EmailNotifierSnapshot {
```
Comment (md2119) [inline proto/storage/report_snapshot.proto:45]: notifier name?

**Review by md2119 — APPROVED** [review]

File: proto/storage/report_snapshot.proto (hunk 2 above)
Comment (charmik-redhat) [reply proto/storage/report_snapshot.proto:46]: Added

File: central/reports/snapshot/datastore/datastore_impl.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+func (ds *datastoreImpl) AddReportSnapshot(ctx context.Context, snap *storage.ReportSnapshot) error {
+	defer metrics.SetDatastoreFunctionDuration(time.Now(), resourceType, "AddReportSnapshot")
+	if err := sac.VerifyAuthzOK(workflowSAC.WriteAllowed(ctx)); err != nil {
+		return err
+	}
+	if snap.ReportId == "" {
+		return errors.New("New report snapshot must have a non-empty report id")
+	}
+	snap.ReportId = uuid.NewV4().String()
```
Comment (dashrews78) [inline central/reports/snapshot/datastore/datastore_impl.go:110]: this overwrites the `snap.ReportId` we just required to be not empty.  That seems odd, if that is intentional a comment explaining would be useful.

Comment (charmik-redhat) [reply central/reports/snapshot/datastore/datastore_impl.go:110]: Good catch, that should be removed.

**Pull request merged**